README.md

```diff
@@ -79,14 +51,54 @@ Requires `rust` and `cargo`: [installation instructions.](https://www.rust-lang.
 $ cargo build [--release]
 ```
 
+### Test
+
+```
+$ cargo test --release
+```
+
 ### Example Network Simulation
 
 A basic [example](https://github.com/poanetwork/hbbft/blob/master/examples/README.md) is included to run a network simulation.
 
 ```
-$ cargo run --example simulation -- -h
+$ cargo run --example simulation --release -- -h
 ```
 
+## Current TODOs
+
+See [Issues](https://github.com/poanetwork/hbbft/issues) for all tasks in progress. 
+
+- [ ] Honey Badger encryption ([#41](https://github.com/poanetwork/hbbft/issues/41))
+
+- [ ] Dynamic Honey Badger (adding and removing nodes in a live network environment) ([#47](https://github.com/poanetwork/hbbft/issues/47#issuecomment-394640406))
+
+- [ ] Optimize Coin algorithm using a schedule ([#69](https://github.com/poanetwork/hbbft/issues/69))
+
+- [ ] Networking example to detail Honey Badger implementation
+
+## Protocol Modifications
+
+Our implementation modifies the protocols described in "[The Honey Badger of BFT Protocols](https://eprint.iacr.org/2016/199.pdf)" in several ways:
+*  We use a [pairing elliptic curve library](https://github.com/ebfull/pairing) to implement pairing-based cryptography rather than Gap Diffie-Hellman groups. 
+* We add a `Terminate` message to the Binary Agreement algorithm. Termination occurs following output, preventing the algorithm from running (or staying in memory) indefinitely. ([#53](https://github.com/poanetwork/hbbft/issues/55))
+*  We add a `Conf` message to the Binary Agreement algorithm. An additional message phase prevents an attack if an adversary controls a network scheduler and a node. ([#37](https://github.com/poanetwork/hbbft/issues/37))
+*  We return additional information from the Subset and Honey Badger algorithms that specifies which node input which data. This allows for identification of potentially malicious nodes.
+* We run a Distributed Key Generation (DKG) protocol which does not require a trusted dealer; nodes collectively generate a secret key. This addresses the problem of single point of failure. See [Distributed Key Generation in the Wild](https://eprint.iacr.org/2012/377.pdf).
```

Comment:
Maybe "`We include`"; "`We run`" may be a bit confusing, because the protocol isn't part of Honey Badger itself (only "Dynamic Honey Badger" will actually run it).